libvast/src/system/type_registry.cpp

```diff
@@ -77,18 +77,23 @@ caf::error type_registry_state::save_to_disk() const {
 
 caf::error type_registry_state::load_from_disk() {
   // Nothing to load is not an error.
-  if (!exists(dir)) {
+  std::error_code err{};
```

Comment:
Was this picked up via ADL? My goodness.

Reply:
`dir` is of type `std::filesystem::path`, so what else would you expect?

Reply:
No surprise we didn't catch this one during review.

Reply:
Gotta love ADL. 😒

Reply:
Ouch. Sorry I missed this when I introduced this change 😞 

Thanks for the fix.

Reply:
No worries, these things happen.

Reply:
This is one of those things that are only obvious in hindsight. We probably would have run into the exact same issue.

Reply:
Pedantically, can you help me understand why `exists` unqualified would pick up `std::filesystem::exists`? I don't see us do a `using namespace std;` or anything like that. Who's bringing it into scope?

Reply:
See this recent article on (among other things) ADL: https://preshing.com/20210315/how-cpp-resolves-a-function-call/